app-tracking-protection/vpn-api/src/main/java/com/duckduckgo/mobile/android/vpn/state/VpnStateMonitor.kt

```diff
@@ -56,17 +57,18 @@ interface VpnStateMonitor {
     }
 
     sealed class VpnRunningState {
-        object ENABLING : VpnRunningState()
-        object ENABLED : VpnRunningState()
-        data class DISABLED(val snoozedTriggerAtMillis: Long? = null) : VpnRunningState()
-        object INVALID : VpnRunningState()
+        data object ENABLING : VpnRunningState()
+        data object ENABLED : VpnRunningState()
+        data object DISABLED : VpnRunningState()
+        data object INVALID : VpnRunningState()
     }
 
-    enum class VpnStopReason {
-        SELF_STOP,
-        ERROR,
-        REVOKED,
-        UNKNOWN,
-        RESTART,
+    sealed class VpnStopReason {
+        data object SELF_STOP : VpnStopReason()
+        data object ERROR : VpnStopReason()
+        data object REVOKED : VpnStopReason()
+        data object UNKNOWN : VpnStopReason()
+        data object RESTART : VpnStopReason()
+        data class SNOOZED(val snoozedTriggerAtMillis: Long? = null) : VpnStopReason()
```

Comment:
Why not using SELF_STOP? I am not sure we should introduced a new SNOOZE state. 
Snooze is just self stop that will re-enable itself later and there's a lot of logic that we should re-use for self-stop in a lot of classes

Reply:
It seemed simpler to introduce a new reason as SELF_STOP is used/checked in a bunch of places where we don’t want SNOOZE to be considered. I will see if reusing SELF_STOP will work.